### PR TITLE
Take pipeline out of run fragment, use fields on run directly

### PR DESCRIPTION
**Pull request by gibsondan**
Don't query pipeline snapshot at all when rendering the runs page / gantt chart

### CODE REVIEW

**Comment by gibsondan** [comment]
Current dependencies on/for this PR:
* master
  * **PR #6059** 
    * **PR #6062**  (this PR)

This comment was auto-generated by [Graphite](https://graphite.dev?utm_source=gh).

File: js_modules/dagit/packages/core/src/runs/Run.tsx [hunk 1]
```diff
@@ -231,7 +231,7 @@ const RunWithData: React.FunctionComponent<RunWithDataProps> = ({
     : [];
 
   const onLaunch = async (style: ReExecutionStyle) => {
-    if (!run || run.pipeline.__typename === 'UnknownPipeline' || !repoMatch) {
+    if (!run || !repoMatch) {
```
Comment (prha) [inline js_modules/dagit/packages/core/src/runs/Run.tsx:234]: check `run.pipelineSnapshotId` here?

File: js_modules/dagit/packages/core/src/runs/RunActionButtons.tsx [hunk 2]
```diff
@@ -251,14 +251,6 @@ function usePipelineAvailabilityErrorForRun(
     return null;
   }
```
Comment (prha) [inline js_modules/dagit/packages/core/src/runs/RunActionButtons.tsx:253]:
```
if (run && !run.pipelineSnapshotId) {
   ...
}
```

File: js_modules/dagit/packages/core/src/runs/RunRoot.tsx [hunk 3]
```diff
@@ -130,13 +130,6 @@ const RUN_ROOT_QUERY = gql`
...
       ... on Run {
         id
-        pipeline {
-          __typename
-          ... on PipelineReference {
-            name
-            solidSelection
-          }
-        }
         ...RunFragment
```
Comment (prha) [inline js_modules/dagit/packages/core/src/runs/RunRoot.tsx:133]: just for hygiene, can we add `pipelineName` here outside of the RunFragment splat?

File: js_modules/dagit/packages/core/src/runs/RunUtils.tsx [hunk 4]
```diff
@@ -122,7 +122,7 @@ export function getReexecutionVariables(input: {
 }) {
   const {run, style, repositoryLocationName, repositoryName} = input;
 
-  if (!run || ('pipeline' in run && run.pipeline.__typename === 'UnknownPipeline')) {
+  if (!run) {
```
Comment (prha) [inline js_modules/dagit/packages/core/src/runs/RunUtils.tsx:125]:
```
if (!run || !run.pipelineSnapshotId) {
    ...
}
```

File: js_modules/dagit/packages/core/src/runs/RunFragments.tsx [hunk 5]
```diff
@@ -36,13 +36,7 @@ export const RunFragments = {
...
       parentRunId
       pipelineName
-      pipeline {
-        __typename
-        ... on PipelineReference {
-          name
-          solidSelection
-        }
-      }
+      solidSelection
```
Comment (prha) [inline js_modules/dagit/packages/core/src/runs/RunFragments.tsx:39]: pipelineSnapshotId?

**Review by prha — APPROVED** [review]

**Pull request merged**